### PR TITLE
ci: Update cicd for cloudsql

### PR DESCRIPTION
**Pull request by Yuan325**

### CODE REVIEW

File: retrieval_service/example-config-cloudsql.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 host: 0.0.0.0
 datastore:
   # Example for Cloud SQL
-  kind: "cloudsql-postgres"
+  kind: "cloudsql-dialect"
```
Comment (kurtisvg) [inline retrieval_service/example-config-cloudsql.yml:4]: Why is this dialect and not a specific engine type? They should still be different datastores right?

File: retrieval_service/datastore/providers/alloydb.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ class Client(datastore.Client[Config]):
 
     @datastore.classproperty
     def kind(cls):
-        return "alloydb-postgres"
+        return POSTGRES_IDENTIFIER
```
Comment (kurtisvg) [inline retrieval_service/datastore/providers/alloydb.py:49]: nit: Should probably be "ALLOYDB_PG_IDENTIFIER"

File: retrieval_service/datastore/providers/cloudsql_postgres.py [hunk 3]
```diff
@@ -45,7 +45,7 @@ class Client(datastore.Client[Config]):
 
     @datastore.classproperty
     def kind(cls):
-        return "cloudsql-postgres"
+        return POSTGRES_IDENTIFIER
```
Comment (kurtisvg) [inline retrieval_service/datastore/providers/cloudsql_postgres.py:48]: nit: CLOUD_SQL_PG_IDENTIFIER

File: retrieval_service/datastore/providers/firestore.py [hunk 4]
```diff
@@ -36,7 +38,7 @@ class Client(datastore.Client[Config]):
 
     @datastore.classproperty
     def kind(cls):
-        return "firestore"
+        return DB_IDENTIFIER
```
Comment (kurtisvg) [inline retrieval_service/datastore/providers/firestore.py:41]:
```suggestion
        return FIRESTORE_IDENTIFIER
```

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: retrieval_service/example-config-cloudsql.yml (hunk 1 above)
Comment (Yuan325) [reply retrieval_service/example-config-cloudsql.yml:4]: Update this to be `cloudsql-engine`. The other configs is the same across cloudsql-postgres and cloudsql-mysql so I was thinking we can reduce the duplication of files by using the same ones. This is used during cloudbuild workflow to setup config.yml.

File: retrieval_service/datastore/providers/cloudsql_postgres_test.py [hunk 5]
```diff
@@ -40,12 +40,12 @@
 
 @pytest.fixture(scope="module")
 def db_user() -> str:
-    return get_env_var("DB_USER", "name of a postgres user")
+    return get_env_var("DB_USER_PG", "name of a postgres user")
```
Comment (kurtisvg) [inline retrieval_service/datastore/providers/cloudsql_postgres_test.py:43]: do you still want to change this?

**Review by kurtisvg — APPROVED** [review]

**Pull request merged**